lib/actions-oclif/ssh.ts

```diff
@@ -0,0 +1,390 @@
+/**
+ * @license
+ * Copyright 2016-2020 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { flags } from '@oclif/command';
+import { stripIndent } from 'common-tags';
+import Command from '../command';
+import * as cf from '../utils/common-flags';
+import { getBalenaSdk } from '../utils/lazy';
+import {
+	parseAsInteger,
+	validateDotLocalUrl,
+	validateIPAddress,
+} from '../utils/validation';
+import * as BalenaSdk from 'balena-sdk';
+
+interface FlagsDef {
+	port?: number;
+	tty: boolean;
+	verbose: boolean;
+	noproxy: boolean;
+	help: void;
+}
+
+interface ArgsDef {
+	applicationOrDevice: string;
+	serviceName?: string;
+}
+
+export default class NoteCmd extends Command {
+	public static description = stripIndent`
+		SSH into the host or application container of a device.
+
+		Start a shell on a local or remote device. If a service name is not provided,
+		a shell will be opened on the host OS.
+
+		If an application name is provided, an interactive menu will be presented
+		for the selection of an online device. A shell will then be opened for the
+		host OS or service container of the chosen device.
+
+		For local devices, the IP address and .local domain name are supported.
+		If the device is referenced by IP or \`.local\` address, the connection
+		is initiated directly to balenaOS on port \`22222\` via an
+		openssh-compatible client. Otherwise, any connection initiated remotely
+		traverses the balenaCloud VPN.
+
+		Commands may be piped to the shell, for example:
+			\`echo "uptime; exit;" | ./bin/balena ssh f49cefd\`
+
+		Note: \`balena ssh\` requires an openssh-compatible client to be correctly
+		installed in your shell environment. For more information (including Windows
+		support) please check:
+			https://github.com/balena-io/balena-cli/blob/master/INSTALL.md#additional-dependencies,
+	`;
+
+	public static examples = [
+		'$ balena ssh MyApp',
+		'$ balena ssh f49cefd',
+		'$ balena ssh f49cefd my-service',
+		'$ balena ssh f49cefd --port <port>',
+		'$ balena ssh 192.168.0.1 --verbose',
+		'$ balena ssh f49cefd.local my-service',
+	];
+
+	public static args = [
+		{
+			name: 'applicationOrDevice',
+			description: 'application name, device uuid, or address of local device',
+			required: true,
+		},
+		{
+			name: 'serviceName',
+			description: 'service name, if connecting to container',
```

Comment:
`"to a container"` would read a bit better to me

Reply:
ok sure

Reply:
It's acceptable to drop articles in English when space is at a premium (e.g. in newspaper headlines), but no need here.